tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -83,6 +83,199 @@ public function data_provider_delayed_strategies() {
 		);
 	}
 
+	/**
+	 * Data provider for test_print_delayed_inline_script_loader_timing.
+	 *
+	 * @return array[]
+	 */
+	public function data_to_test_print_delayed_inline_script_loader_timing() {
+		/**
+		 * Enqueue script with defer strategy.
+		 *
+		 * @param bool $in_footer In footer.
+		 */
+		$enqueue_script    = static function ( $in_footer = false ) {
+			wp_enqueue_script(
+				$in_footer ? 'foo-footer' : 'foo-head',
+				sprintf( 'https://example.com/%s.js', $in_footer ? 'foo-footer' : 'foo-head' ),
+				array(),
+				null,
+				array(
+					'in_footer' => $in_footer,
+					'strategy'  => 'defer',
+				)
+			);
+		};
+
+		/**
+		 * Add inline after script.
+		 *
+		 * @param string $handle Handle.
+		 */
+		$add_inline_script = static function ( $handle ) {
+			wp_add_inline_script(
+				$handle,
+				"/*{$handle}-after*/"
+			);
+		};
+
+		return array(
+			'no_delayed_inline_scripts'            => array(
+				'set_up'          => static function () use ( $enqueue_script ) {
+					$enqueue_script( false );
+					$enqueue_script( true );
+				},
+				'expected_head'   => <<<HTML
+<script id="foo-head-js" src="https://example.com/foo-head.js" type="text/javascript" defer></script>
+HTML
+				,
+				'expected_torso'  => '',
+				'expected_footer' => <<<HTML
+<script id="foo-footer-js" src="https://example.com/foo-footer.js" type="text/javascript" defer></script>
+HTML
+				,
+			),
+			'delayed_inline_script_in_head_only'   => array(
+				'set_up'          => static function () use ( $enqueue_script, $add_inline_script ) {
+					$enqueue_script( false );
+					$add_inline_script( 'foo-head' );
+				},
+				'expected_head'   => $this->get_delayed_inline_script_loader_script_tag() . <<<HTML
+<script id="foo-head-js" src="https://example.com/foo-head.js" type="text/javascript" defer></script>
+<script id="foo-head-js-after" type="text/plain">
+/*foo-head-after*/
+</script>
+HTML
+				,
+				'expected_torso'  => '',
+				'expected_footer' => '',
+			),
+			'delayed_inline_script_in_footer_only' => array(
+				'set_up'          => static function () use ( $enqueue_script, $add_inline_script ) {
+					$enqueue_script( true );
+					$add_inline_script( 'foo-footer' );
+				},
+				'expected_head'   => $this->get_delayed_inline_script_loader_script_tag(), // TODO: This script is getting output even though it isn't needed yet.
```

Comment:
The fact that the delayed loader script is being printed in the `wp_head` even though all the scripts are enqueued for `wp_footer`, I don't think this is a big problem. True it means that the script is listening for more `load` events than it has to. But to address this we'd have to somehow extend `WP_Scripts::has_delayed_inline_script()` to be contextually aware of _when_ in the document it is being called. In other words, if `doing_action('wp_head')` then we'd want to ignore any scripts that were registered as targeting `in_footer`.